### PR TITLE
Feature/farm 318 take picture

### PR DESCRIPTION
**Pull request by davidfranquet**
[FARM-318]

@LeeMatthewHiggins Can you test it on you iPhone? 😄 

#### 📲 What

Helper to take an image from camera or gallery and return a file

#### 🤔 Why
		
Needed for update the image profile 
		
#### 👀 See

ezgif com-video-to-gif-2
ezgif com-video-to-gif

		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -41,5 +41,10 @@
 	</array>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<false/>
+	<key>NSPhotoLibraryUsageDescription</key>
+	<string>Take profile picture</string>
+	<key>NSCameraUsageDescription</key>
+	<string>Take profile picture</string>
+	<key>NSMicrophoneUsageDescription</key>
```
Comment (LeeMatthewHiggins) [inline ios/Runner/Info.plist]: might need to fill that out

File: lib/ui/common/take_image.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  final ImageSource imageSource;
+  final int imageMaxHeight;
+  final int imageMaxWidth;
+  final void Function(File) onImageTaken;
+  final bool circleShapeOnCrop;
+  final double imageRatioX;
+  final double imageRatioY;
+
+
+  TakeImage({
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/take_image.dart]:
```suggestion
  ImagePicker({
```

**Review by albertmiro — COMMENTED** [review]
Nice GIF in the PR description! :-D

File: lib/ui/playground/playground_take_image_tester.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+class _PlaygroundTakeImageTesterState extends State<PlaygroundTakeImageTester> {
+  File _file;
+
+  @override
+  void initState() {
+    super.initState();
+
+    TakeImage(
```
Comment (LeeMatthewHiggins) [inline lib/ui/playground/playground_take_image_tester.dart]: what about a static helper that turn the use into a one liner
ImagePicker.pickImage( success { }, onError { });

File: lib/ui/playground/data/playground_tasks_datasource.dart [hunk 4]
```diff
@@ -246,30 +242,39 @@ class PlayGroundTasksDataSource implements PlaygroundDataSource {
...
-                  engine: _engine,
-                ))),]))
+      PlaygroundWidget(
+        title: 'TASK FARM-97 Update Recomentations',
+        child: PlaygroundView(widgetList: [
+          PlaygroundWidget(
+              title: "TASK FARM-97 View List states",
+              child: RecommendationsList(
+                  provider: SequencedViewModelProvider<
+                      RecommendationsListViewModel>(
```
Comment (albertmiro) [inline lib/ui/playground/data/playground_tasks_datasource.dart]: Can we move the `<` to the next line?

File: ios/Runner/Info.plist (hunk 1 above)
Comment (davidfranquet) [reply ios/Runner/Info.plist]: Yes we don't need the mic

File: lib/ui/common/take_image.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/ui/common/take_image.dart]: The lib that I'm using is called ImagePicker. I'll modify the import as this: 
`import 'package:image_picker/image_picker.dart' as ImagePickerLib
    show ImagePicker, ImageSource;`
 to be able to use ImagePicker on this class

File: lib/ui/playground/playground_take_image_tester.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/ui/playground/playground_take_image_tester.dart]: Yep much better

File: lib/ui/playground/data/playground_tasks_datasource.dart (hunk 4 above)
Comment (davidfranquet) [reply lib/ui/playground/data/playground_tasks_datasource.dart]: DONE. 
Added missing commas to format it correctly

File: lib/ui/playground/playground_take_image_tester.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/ui/playground/playground_take_image_tester.dart]: Done

File: lib/ui/common/take_image.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/ui/common/take_image.dart]: Done

File: ios/Runner/Info.plist (hunk 1 above)
Comment (davidfranquet) [reply ios/Runner/Info.plist]: Done

File: lib/ui/common/image_picker.dart [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+class _Strings {
+  static final editPhoto = 'Edit Photo';
+  static final noImagePickedError = 'No image picked';
+  static final noCroppedImageError = 'No cropped image';
+}
+
+class ImagePicker {
+  static Future<void> pickImage({
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/image_picker.dart]: could return if image was picked or not?

**Comment by davidfranquet** [comment]
@LeeMatthewHiggins can you pull this branch and test it on your iPhone?

File: lib/ui/common/image_picker.dart (hunk 5 above)
Comment (davidfranquet) [reply lib/ui/common/image_picker.dart]: 👍

Comment (davidfranquet) [reply lib/ui/common/image_picker.dart]: Done

File: lib/ui/common/image_picker.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:image_cropper/image_cropper.dart';
+import 'package:image_picker/image_picker.dart' as ImagePickerLib
+    show ImagePicker, ImageSource;
```
Comment (albertmiro) [inline lib/ui/common/image_picker.dart:6]: Nice! I didn't know that!

File: lib/ui/common/image_picker.dart [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    @required int imageMaxWidth,
+    bool circleShapeOnCrop = true,
+    double imageRatioX = _Constants.defaultImageRatioX,
+    double imageRatioY = _Constants.defaultImageRatioY,
+  }) async {
+    final file =
+        await ImagePickerLib.ImagePicker.pickImage(source: imageSource);
+
+    if (file == null) {
+      onError(_Strings.noImagePickedError);
```
Comment (albertmiro) [inline lib/ui/common/image_picker.dart]: Not sure about this comment, I've seen in another PR's that text is wrapped with:

```
Intl.message(_Strings.noImagePickedError)
```

To be able to be translated to other languages, that should be done in this way?

File: lib/ui/common/image_picker.dart [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      statusBarColor: _Constants.cropStatusBarColor,
+      toolbarColor: _Constants.cropToolbarBarColor,
+      toolbarTitle: Intl.message(_Strings.editPhoto),
+      toolbarWidgetColor: Colors.black,
+    );
+
+    file.delete();
+
+    if (croppedFile == null) {
+      onError(_Strings.noCroppedImageError);
```
Comment (albertmiro) [inline lib/ui/common/image_picker.dart]: Wrap the text with `Intl.message()`

File: lib/ui/playground/playground_take_image_tester.dart [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  @override
+  void dispose() {
+    super.dispose();
+    _file?.delete();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: _file != null ? Image.file(_file) : Text(_Strings.noImage),
```
Comment (albertmiro) [inline lib/ui/playground/playground_take_image_tester.dart:57]: Wrap the text with `Intl.message()`

File: lib/ui/common/image_picker.dart (hunk 7 above)
Comment (davidfranquet) [reply lib/ui/common/image_picker.dart]: yep you're right

Comment (davidfranquet) [reply lib/ui/common/image_picker.dart]: done

File: lib/ui/common/image_picker.dart (hunk 8 above)
Comment (davidfranquet) [reply lib/ui/common/image_picker.dart]: done

File: lib/ui/playground/playground_take_image_tester.dart (hunk 9 above)
Comment (davidfranquet) [reply lib/ui/playground/playground_take_image_tester.dart:57]: this is for testing area (playground)

File: lib/ui/playground/data/playground_tasks_datasource.dart [hunk 10]
```diff
@@ -1,45 +1,41 @@
 import 'package:farmsmart_flutter/model/bloc/SequencedViewModelProvider.dart';
```
Comment (LeeMatthewHiggins) [inline lib/ui/playground/data/playground_tasks_datasource.dart:1]: this file in on my hit list

**Review by LeeMatthewHiggins — DISMISSED** [review]
LGTM

**Review by albertmiro — DISMISSED** [review]
🚀

**Review by albertmiro — APPROVED** [review]

**Review by LeeMatthewHiggins — APPROVED** [review]

**Pull request merged**